src/ModalFooter.js

```diff
@@ -1,31 +1,32 @@
 import classNames from 'classnames';
 import React from 'react';
+import PropTypes from 'prop-types';
 import elementType from 'prop-types-extra/lib/elementType';
 
-import { bsClass, getClassSet, splitBsProps } from './utils/bootstrapUtils';
+import { createBootstrapComponent } from './ThemeProvider';
 
-const propTypes = {
-  componentClass: elementType
-};
+class ModalFooter extends React.Component {
+  static propTypes = {
+    bsPrefix: PropTypes.string,
+    componentClass: elementType
+  };
 
-const defaultProps = {
-  componentClass: 'div'
-};
+  static defaultProps = {
+    componentClass: 'div'
+  };
 
-class ModalFooter extends React.Component {
   render() {
-    const { componentClass: Component, className, ...props } = this.props;
-    const [bsProps, elementProps] = splitBsProps(props);
+    const {
+      componentClass: Component,
+      bsPrefix,
+      className,
+      ...props
+    } = this.props;
 
-    const classes = getClassSet(bsProps);
+    const classes = classNames(className, bsPrefix);
 
-    return (
-      <Component {...elementProps} className={classNames(className, classes)} />
-    );
+    return <Component {...props} className={classNames(classes)} />;
```

Comment:
this all looks great generally. i'd remove the extra `classNames` call here and right above, you can combine that all into `className={classNames(className, bsPrefix)}`